Introduce pull request checklist (#7329)
DataDog/dd-trace-java: .github/pull_request_template.md

# Additional Notes

# Contributor Checklist

- [ ] Format the title [according the contribution guidelines](/CONTRIBUTING.md#title-format)
- [ ] Assign the `type:` and (`comp:` or `inst:`) labels in addition to [any usefull labels](/CONTRIBUTING.md#labels)

> - [ ] Squash your commits prior merging

Reviewer: ```suggestion
- [ ] Squash your commits prior merging or merge using GitHub's [Squash and merge](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/about-pull-request-merges#squash-and-merge-your-commits)
```

> - [ ] Don't use `close`, `fix` or any [linking keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) when referencing an issue,  
>   but use `solves` instead, and assign the PR milestone to the issue

Reviewer: ```suggestion
- [ ] Don't use `close`, `fix` or any [linking keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) when referencing an issue.
  Use `solves` instead, and assign the PR [milestone](https://github.com/DataDog/dd-trace-java/milestones) to the issue
```

# Additional Notes

# Contributor Checklist

> - [ ] Format the PR title to describe the changes:

Reviewer: Very nit-picky, but the trailing `:`, `,` and `.` are not needed and implied by the checklist structure

  - Using a sentence with an infinitve verb,
  - In a way that makes sense for the users,

>   - Avoiding ASCII tag (`[PROD]`) or prefix (`PROD - `).

Reviewer: ```suggestion
  - Avoiding prefixes (`[PROD]`) or prefix (`PROD - `) in favor of labels.
```

- [ ] Assign the appropriate labels:

>   - One to describe the kind of changes from the `type:` section,

Reviewer: ```suggestion
  - [Required] To describe the kind of changes from the `type:` section
```
